src/datastore/pg-store.ts

```diff
@@ -448,42 +448,39 @@ export class PgStore {
       const countQuery = await sql<
         { total: number }[]
       >`SELECT microblock_count AS total FROM chain_tip`;
-      const microblockQuery = await sql<
-        (MicroblockQueryResult & { tx_id?: string | null; tx_index?: number | null })[]
-      >`
-        SELECT microblocks.*, tx_id FROM (
-          SELECT ${sql(MICROBLOCK_COLUMNS)}
-          FROM microblocks
-          WHERE canonical = true AND microblock_canonical = true
-          ORDER BY block_height DESC, microblock_sequence DESC
-          LIMIT ${args.limit}
-          OFFSET ${args.offset}
-        ) microblocks
-        LEFT JOIN (
-          SELECT tx_id, tx_index, microblock_hash
-          FROM txs
-          WHERE canonical = true AND microblock_canonical = true
-          ORDER BY tx_index DESC
-        ) txs
```

Comment:
What's inefficient about this join here is that it is first creating a *huge* table of all canonical txs sorted by `tx_index` and then filters it on `microblock_hash`.

You shouldn't need to separate this into several queries, instead, try to perform the JOIN *without* sub-queries:
```
SELECT ... FROM microblocks
LEFT JOIN txs USING (microblock_hash)
WHERE ...
```